code/game/mecha/mecha.dm

```diff
@@ -440,11 +440,24 @@
 		target = safepick(view(3,target))
 		if(!target)
 			return
+
+	var/mob/living/L = user
+	var/obj/structure/closet/C = target
 	if(!Adjacent(target))
 		if(selected && selected.is_ranged())
+			if(L.has_trait(TRAIT_PACIFISM) && !selected.pacifist_safe)
+				to_chat(user, "<span class='warning'>You don't want to harm other living beings!</span>")
+				return
 			if(selected.action(target,params))
 				selected.start_cooldown()
 	else if(selected && selected.is_melee())
+		if(isliving(target) && !selected.pacifist_safe && L.has_trait(TRAIT_PACIFISM))
+			to_chat(user, "<span class='warning'>You don't want to harm other living beings!</span>")
+			return
+		if(istype(C) && L.has_trait(TRAIT_PACIFISM) && !selected.pacifist_safe)
+			for(var/mob/living/M in C.contents)

```

Comment:
tiny nitpick, you can just do `in C` instead of `in C.contents`

Reply:
alright thanks. Made the change. Also do you know of any better ways to check the contents of a container? Having it do a for loop felt a bit hacky.